### DISCUSSION TITLE
Chunk Error while build  app

### DISCUSSION THREAD

**Discussion by ronnelsuan134** [vercel/next.js#62546] [Help]
### Summary

Is there someone who can explain to me why my Next.js build is behaving like this?

My Next.js application renders the correct page initially but shown JSON data when i refresh the page.
![Screenshot 2024-02-26 233437](https://github.com/vercel/next.js/assets/82152144/4d0427ea-12b4-4172-bcf7-b6e9d6d8f80b)

**Comment by icyJoseph** [comment]
It is a problem with your CDN:

- https://github.com/vercel/next.js/issues/48569#issuecomment-1514982815

When you soft navigate onto this page, an SPA transition happens, where the props for the page are fetched using the same path, with an `rsc` query, IIRC, and that is getting stored in your CDN's cache. Next time you reload the page, you get served the raw RSC from the CDN, instead of hitting the server to generate/get the HTML for the page.

**Reply by ronnelsuan134** [reply]
what's the possible solution? i'm using AWS CF for my CDN

**Reply by icyJoseph** [reply]
Right, in that link I posted, there's, I think, a way to do this in CloudFront. You just need to make sure that the CDN uses the Vary Header in the request.

**Reply by icyJoseph** [reply]
I found something here,

- https://github.com/vercel/next.js/issues/49130#issuecomment-1639058830

**Reply by ronnelsuan134** [reply]
thank you for the help :)

**Reply by icyJoseph** [reply]
Did you get it to work?
